main/src/library/LazyList.flix

```diff
@@ -74,202 +64,171 @@ namespace LazyList {
     /// Does not force the tail of the list `xs`.
     ///
     pub def head(xs: LazyList[a]): Option[a] = match xs {
-        case Empty => None
-        case LazyCons(h, _) => Some(h)
+        case ENil        => None
+        case ECons(x, _) => Some(x)
+        case LCons(x, _) => Some(x)
+        case LList(ys)   => head(force ys)
     }
 
     ///
-    /// Returns `Some(x)` if `x` is the last element of `xs`.
-    ///
-    /// Returns `None` if `xs` is empty.
+    /// Returns true if and only if `xs` is the empty LazyList, i.e. `ENil`.
     ///
-    /// Forces the entire list `xs`.
+    /// Does not force the tail of the list `xs`.
     ///
-    pub def last(xs: LazyList[a]): Option[a] = match xs {
-        case Empty => None
-        case LazyCons(h, t) => {
-            let tail = force t;
-            if (isEmpty(tail)) Some(h) else last(tail)
-        }
+    pub def isEmpty(xs: LazyList[a]): Bool = match xs {
+        case ENil => true
+        case _    => false
```

Comment:
We'd need to check the `LList` case here too, right?

Reply:
Yes

Reply:
Oops, yeah I see that now. Thanks